internal/git_clone_test.go

```diff
@@ -59,5 +59,5 @@ func TestAccGitCloneRepositoryDoesNotExist(t *testing.T) {
 	tmpDir := t.TempDir()
 
 	err := git.Clone(context.Background(), "doesnot-exist", "", tmpDir)
-	assert.Contains(t, err.Error(), `repository 'https://github.com/databricks/doesnot-exist/' not found`)
+	assert.Error(t, err)
```

Comment:
Now the test does not really do what it intents to do, right?
Why does the original error even occur?

Reply:
To some degree yes. This test now tests that running `git clone` on a repo that does not exist returns an error. 

Fair point though. I'll add a check that this error originates from the git tool. Does that sound good?

Reply:
> This test now tests that running git clone on a repo that does not exist returns an error.

We should expect a specific error though, right now with failed git setup this test does not even exercise code paths responsible for handling non existing repos if I understand it correctly.
If in some time in future we (or git) change the logic on how they handle non existing repos, we won't catch this with this test

Reply:
> test does not even exercise code paths responsible for handling non existing repos
The code for handing non-existing public repos is exercised. What seems to be happening in the current error is:
1. Git tries to resolve the repository and fails. This is because no such public repo exists.
2. Git prompts for a username, which fails because the terminal is not a TTY

The reason I am OK with this approach is with the other unit tests we know that `git clone` works file for existing public repos.

The way to test the `repository does not exist` error would be to set valid git credentials here, which I am not sure how to proceed with

Reply:
Isn't behaviour the same for non existing and private repos in this case? So the test won't really distinguish why we fail

Reply:
I would like to avoid messing with global git config in our git envs though, we can unset this once the test run is over though. IMO the value added by this test is that we fail if the repo does not exist publically and we can defer to the git CLI for the exact error.

I am fine with both approaches though. 

@pietern WDYT?

Reply:
Let's not muck with git configs. That can leak into the user's configuration and is dangerous.

Looking at overrides through env vars (see https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables) you could try setting `GIT_ASKPASS` to `echo` or some other command that immediately returns and I would expect the prompt for credentials to go away.

Reply:
setting `GIT_ASKPASS` does not seem to solve this issue:

runs:
https://github.com/databricks/eng-dev-ecosystem/actions/runs/5678877830/job/15389950952
https://github.com/databricks/eng-dev-ecosystem/actions/runs/5678877036

Reply:
Discussed this offline with @pietern . There's no obvious way to inject fake git credentials to reproduce this error message.

Given the low ROI on trying to set mock user creds, and reasonable confidence that the `git clone` command would return a useful error message, I propose it's fine for us to proceed with just asserting that the error originated from the git CLI